dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceService.java

```diff
@@ -449,6 +457,204 @@ public synchronized void update(LogProviderConfig config) {
         }
     }
 
+    public static void convertStringtoMap(String value) {
+        TraceComponent tc = Tr.register(LogTraceData.class, NLSConstants.GROUP, NLSConstants.LOGGING_NLS);
+
+        Map<String, String> messageMap = new HashMap<>();
+        Map<String, String> traceMap = new HashMap<>();
+        Map<String, String> ffdcMap = new HashMap<>();
+        Map<String, String> accessLogMap = new HashMap<>();
+
+        if (value == null || value == "") {
+            //if no property is set, return
+            return;
+        }
+
+        String[] keyValuePairs = value.split(","); //split the string to create key-value pairs
+        for (String pair : keyValuePairs) //iterate over the pairs
+        {
+            String[] entry = pair.split(":"); //split the pairs to get key and value
+            if (entry.length == 2) {//if the mapped value is intended for all event types
+                //add properties to all the hashmaps and trim whitespaces
+                messageMap.put(entry[0].trim(), entry[1].trim());
+                traceMap.put(entry[0].trim(), entry[1].trim());
+                ffdcMap.put(entry[0].trim(), entry[1].trim());
+                accessLogMap.put(entry[0].trim(), entry[1].trim());
+            } else if (entry.length == 3) {
+                //add properties to their respective hashmaps and trim whitespaces
+                if (CollectorConstants.MESSAGES_LOG_EVENT_TYPE.equals(entry[0].trim())) {
+                    messageMap.put(entry[1].trim(), entry[2].trim());
+                } else if (CollectorConstants.TRACE_LOG_EVENT_TYPE.equals(entry[0])) {
+                    traceMap.put(entry[1].trim(), entry[2].trim());
+                } else if (CollectorConstants.FFDC_EVENT_TYPE.equals(entry[0].trim())) {
+                    ffdcMap.put(entry[1].trim(), entry[2].trim());
+                } else if (CollectorConstants.ACCESS_LOG_EVENT_TYPE.equals(entry[0].trim())) {
+                    accessLogMap.put(entry[1].trim(), entry[2].trim());
+                }
+            } else {
+                Tr.warning(tc, "JSON_FIELDS_FORMAT_WARNING");
+            }
+        }
+        addMapsToNamesArray(messageMap, traceMap, accessLogMap, ffdcMap);
+    }
+
+    public static void addMapsToNamesArray(Map<String, String> messageMap, Map<String, String> traceMap, Map<String, String> accessLogMap, Map<String, String> ffdcMap) {
```

Comment:
need to move this logic into each of the individual event types (or a superclass of the event types)